lib/local-links-manager/distributed_lock.rb

```diff
@@ -2,20 +2,34 @@
 
 module LocalLinksManager
   class DistributedLock
+    attr_accessor :lock_name
+
+    APP = "local-links-manager"
     LIFETIME = (60 * 60) # seconds
 
     def initialize(lock_name)
       @lock_name = lock_name
     end
 
     def lock(lock_obtained:, lock_not_obtained:)
-      Services.redis.lock("local-links-manager:#{Rails.env}:#{@lock_name}", life: LIFETIME) do
+      Services.redis.lock("#{APP}:#{lock_name}", life: LIFETIME, owner: "#{APP}") do
         Rails.logger.debug('Successfully got a lock. Running...')
         lock_obtained.call
       end
     rescue Redis::Lock::LockNotAcquired => e
-      Rails.logger.debug("Failed to get lock for #{@lock_name} (#{e.message}). Another process probably got there first.")
+      Rails.logger.debug("Failed to get lock for #{lock_name} (#{e.message}). Another process probably got there first.")
       lock_not_obtained.call
     end
+
+    def unlock
+      Redis::Lock.new(Services.redis, "#{APP}:#{lock_name}", owner: "#{APP}").unlock
```

Comment:
I guess we're not using `Services.redis.unlock` here because that doesn't let us specify the `owner` param (which is why it wasn't unlocking across processes?).  It'd be useful to mention this in the commit message, or a comment as it's important detail we might otherwise forget and break.

I might even consider changing how we obtain the lock to avoid the `Services.redis.lock` method and go long-hand with `Redis::Lock.new(Services.redis, "#{APP}:#{lock_name}", owner: APP).lock` so that our `lock` and `unlock` methods are more similar.   Perhaps with a comment like: avoid convenience methods on Services.redis to give us more options.  Not hugely convinced by this though, feel free to disregard this idea.